common/inc/com/centreon/common/process.hh

```diff
@@ -0,0 +1,186 @@
+/*
+** Copyright 2024 Centreon
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** For more information : contact@centreon.com
+*/
+#ifndef CENTREON_AGENT_CHECK_PROCESS_HH
+#define CENTREON_AGENT_CHECK_PROCESS_HH
+
+namespace com::centreon::common {
+
+namespace detail {
+// here to limit included files
+struct boost_process;
+}  // namespace detail
+
+/**
+ * @brief This class allow to exec a process asynchronously.
+ * It's a base class. If you want to get stdin and stdout returned data, you
+ * must inherit from this and override on_stdout_read and on_stderr_read
+ * You can call start_process at any moment, if a process is already running,
+ * it's killed
+ * As we can start a process at any moment, all handlers take a caller in
+ * parameter, if this caller is not equal to current _proc, we do nothing.
+ * When handler like on_stdout_read are called, _protect is already locked
```

Comment:
I don't understand this paragraph.
if on_stdout_read is a handler, I don't see the handler in parameter.

Reply:
done